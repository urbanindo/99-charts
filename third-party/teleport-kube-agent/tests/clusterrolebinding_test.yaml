suite: ClusterRoleBinding
templates:
  - clusterrolebinding.yaml
tests:
  - it: creates a ClusterRoleBinding
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ClusterRoleBinding
      - matchSnapshot: {}

  - it: sets ClusterRoleBinding labels when specified
    values:
      - ../.lint/extra-labels.yaml
    asserts:
      - equal:
          path: metadata.labels.app\.kubernetes\.io/name
          value: teleport-kube-agent
      - equal:
          path: metadata.labels.resource
          value: clusterrolebinding
      - matchSnapshot: {}
