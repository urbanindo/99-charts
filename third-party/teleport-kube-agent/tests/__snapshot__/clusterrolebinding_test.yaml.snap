creates a ClusterRoleBinding:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: RELEASE-NAME
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: RELEASE-NAME
    subjects:
    - kind: ServiceAccount
      name: RELEASE-NAME
      namespace: NAMESPACE
sets ClusterRoleBinding labels when specified:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      labels:
        app.kubernetes.io/name: teleport-kube-agent
        resource: clusterrolebinding
      name: RELEASE-NAME
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: RELEASE-NAME
    subjects:
    - kind: ServiceAccount
      name: RELEASE-NAME
      namespace: NAMESPACE
